dispatcher/src/test/java/io/zeebe/dispatcher/integration/DispatcherIntegrationTest.java

```diff
@@ -197,180 +196,6 @@ public void run() {
-              }
-            });
-
-    final AtomicBoolean thread2OvertookThread1 = new AtomicBoolean(false);
-
-    final Thread consumerThread2 =
-        new Thread(
-            () -> {
-              while (consumer2.counter.get() < totalWork) {
-                // in pipeline mode, the second consumer should not overtake the
-                // first consumer
-                if (consumer2.counter.get() > consumer1.counter.get()) {
-                  thread2OvertookThread1.set(true);
-                  // do not leave the loop or else the other thread won't be able to complete
-                }
-
-                subscription2.peekAndConsume(consumer2, Integer.MAX_VALUE);
-              }
-            });
-
-    consumerThread1.start();
-    consumerThread2.start();
-
-    offerMessage(dispatcher, msg, totalWork);
-
-    consumerThread1.join();
-    consumerThread2.join();
-
-    assertThat(thread2OvertookThread1).isFalse();
-
-    dispatcher.close();
-  }
-
-  @Test
-  public void testPeekBlockInPipelineMode() throws Exception {
-    final int totalWork = 1000000;
-    final ClaimedFragment claimedFragment = new ClaimedFragment();
-
-    final BlockPeek blockPeek1 = new BlockPeek();
-    final BlockPeek blockPeek2 = new BlockPeek();
-
-    final Dispatcher dispatcher =
-        Dispatchers.create("default")
-            .actorScheduler(actorSchedulerRule.get())
-            .bufferSize(ByteValue.ofMegabytes(10))
-            .modePipeline()
-            .subscriptions("s1", "s2")
-            .build();
-
-    final Subscription subscription1 = dispatcher.getSubscription("s1");
-    final Subscription subscription2 = dispatcher.getSubscription("s2");
-
-    final AtomicInteger counter1 = new AtomicInteger(0);
-    final AtomicInteger counter2 = new AtomicInteger(0);
-
-    final Thread consumerThread1 =
-        new Thread(
-            () -> {
-              while (counter1.get() < totalWork) {
-                while (subscription1.peekBlock(blockPeek1, alignedFramedLength(64), false) == 0) {}
-
-                counter1.incrementAndGet();
-                blockPeek1.markCompleted();
-              }
-            });
-
-    final Thread consumerThread2 =
-        new Thread(
-            () -> {
-              while (counter2.get() < totalWork) {
-                // in pipeline mode, the second consumer should not overtake the first consumer
-                assertThat(counter2.get()).isLessThanOrEqualTo(counter1.get());
-
-                while (subscription2.peekBlock(blockPeek2, alignedFramedLength(64), false) == 0) {}
-
-                counter2.incrementAndGet();
-                blockPeek2.markCompleted();
-              }
-            });
-
-    consumerThread1.start();
-    consumerThread2.start();
-
-    claimFragment(dispatcher, claimedFragment, totalWork);
-
-    consumerThread1.join();
-    consumerThread2.join();
-
-    dispatcher.close();
-  }
-
-  @Test
-  public void testMarkFragmentAsFailed() throws Exception {
```

Comment:
Probably we could also remove this feature.

Reply:
It seems that it is currently used in transport.

Reply:
Ah ok thanks